docs/tutorials/tabular_prediction/tabular-custom-model.md

```diff
@@ -0,0 +1,374 @@
+# Adding a custom model to AutoGluon
+:label:`sec_tabularcustommodel`
+
+**Tip**: If you are new to AutoGluon, review :ref:`sec_tabularquick` to learn the basics of the AutoGluon API.
+
+This tutorial describes how to add a custom model to AutoGluon that can be trained and ensembled alongside the default models ([default model documentation](../../api/autogluon.tabular.models.html#module-autogluon.tabular.models)).
+
+In this example, we create a custom Random Forest model for use in AutoGluon. All models in AutoGluon inherit from the AbstractModel class ([AbstractModel source code](../../_modules/autogluon/core/models/abstract/abstract_model.html)), and must follow its API to work alongside other models.
+
+Note that while this tutorial provides a basic model implementation, this does not cover many aspects that are used in most implemented models.
+
+To best understand how to implement more advanced functionality, refer to the [source code](../../api/autogluon.tabular.models.html#module-autogluon.tabular.models) of the following models:
+
+- Respecting time limit / early stopping logic
+  - Refer to [LGBModel](../../_modules/autogluon/tabular/models/lgb/lgb_model.html#LGBModel) and [RandomForestModel](../../_modules/autogluon/tabular/models/rf/rf_model.html#RFModel)
+- Respecting memory usage limit
+  - Refer to LGBModel and RandomForestModel
+- Sample weight support
+  - Refer to LGBModel
+- Validation data and eval_metric usage
+  - Refer to LGBModel
+- GPU training support
+  - Refer to LGBModel
+- Save / load logic of non-serializable models
+  - Refer to [NNFastAiTabularModel](../../_modules/autogluon/tabular/models/fastainn/tabular_nn_fastai.html#NNFastAiTabularModel)
+- Advanced problem type support (Softclass, Quantile)
+  - Refer to RandomForestModel
+- Text feature type support
+  - Refer to [TextPredictorModel](../../_modules/autogluon/tabular/models/text_prediction/text_prediction_v1_model.html#TextPredictorModel)
+- Image feature type support
+  - Refer to [ImagePredictorModel](../../_modules/autogluon/tabular/models/image_prediction/image_predictor.html#ImagePredictorModel)
+- Custom HPO logic
+  - Refer to LGBModel
+
```

Comment:
Can we make this a table?

aspect | reference implementation
-------|--------------------------
Respecting time limit / early stopping logic | [LGBModel](../../_modules/autogluon/tabular/models/lgb/lgb_model.html#LGBModel) and [RandomForestModel](../../_modules/autogluon/tabular/models/rf/rf_model.html#RFModel)

Reply:
Do you know how to do that in the docs?

Reply:
Agreed the current bullet pt formatting is confusing because nested bullets appear to be part of sequential bulleted list in the docs.  Could alternatively just reformat it like this: 

Respecting time limit / early stopping logic:  Refer to LGBModel
Respecting memory usage limit:  Refer to RandomForestModel
...

Reply:
Something is weird with D2L's formatting... I added a table but it also has issues with spacing that it shouldn't. Similarly, the nested bullets look fine in my PyCharm preview of markdown, but don't work in our built docs.

Reply:
I'll merge for now, but we should really find a better way to do these things... I don't understand why what we are using has less working functionality than default markdown.